additional_data_sources/siren-function-source/scripts/import_siren.py

```diff
@@ -0,0 +1,192 @@
+from datetime import datetime, timedelta
+import pandas as pd
+import requests
+import time
+from scripts.utils import GCP_PROJECT, BIGQUERY_CLEAN_DATASET, BUCKET_NAME, TOKEN
+
+SIREN_FILENAME = "siren_data.csv"
+LIME_DATE_SIREN = 35
+MAX_SIREN_CALL = 150
+
+
+def get_limit_date():
+    limit_date = (datetime.now() - timedelta(days=LIME_DATE_SIREN)).strftime("%Y-%m-%d")
+    return limit_date
+
+
+def get_offerer_siren_list():
+    return pd.read_gbq(
+        f"""SELECT offerer_siren as siren
+        FROM `{GCP_PROJECT}.{BIGQUERY_CLEAN_DATASET}.applicative_database_offerer` 
+        WHERE offerer_siren is not null
+        """
+    )
+
+
+def get_siren_query(siren_list):
+    query = "https://api.insee.fr/entreprises/sirene/V3/siren?q="
+    for i in range(len(siren_list) - 1):
+        query += "".join(f"""siren:{siren_list[i]} OR """)
+    query += f"""siren:{siren_list[len(siren_list)-1]}&curseur=*&nombre=1000"""
+    return query
+
+
+def append_info_siren_list(siren_info_list, result):
+    for i in range(len(result["unitesLegales"])):
```

Comment:
plus simple si tu fais :
`for uniteLegale in result["unitesLegales"]:`
et après:
`"siren": uniteLegale["siren"]`